src/main/java/nextstep/subway/application/LineService.java

```diff
@@ -0,0 +1,84 @@
+package nextstep.subway.application;
+
+import nextstep.subway.domain.Line;
+import nextstep.subway.domain.Station;
+import nextstep.subway.dto.LineRequest;
+import nextstep.subway.dto.LineResponse;
+import nextstep.subway.exception.EntityNotFoundException;
+import nextstep.subway.repository.LineRepository;
+import nextstep.subway.repository.StationRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+
+@Service
+public class LineService {
+
+    private LineRepository lineRepository;
+
+    private StationRepository stationRepository;
+
+    public LineService(LineRepository lineRepository,
+                       StationRepository stationRepository) {
+        this.lineRepository = lineRepository;
+        this.stationRepository = stationRepository;
+    }
+
+    @Transactional
+    public LineResponse saveLine(LineRequest lineRequest) {
+        Station upStation = getStation(lineRequest.getUpStationId());
+        Station downStation = getStation(lineRequest.getDownStationId());
+        Line line = lineRepository.save(lineRequest.toLine(upStation, downStation));
+        return LineResponse.of(line);
+    }
+
+    @Transactional(readOnly = true)
+    public LineResponse findById(Long id) {
+        return LineResponse.of(getLine(id));
+    }
+
+    @Transactional(readOnly = true)
+    public List<LineResponse> findAllLines() {
+        return lineRepository.findAll()
+                .stream()
+                .map(LineResponse::of)
+                .collect(Collectors.toList());
+    }
+
+    @Transactional
+    public LineResponse updateById(Long id, LineRequest lineRequest) {
+        Line line = getLine(id);
+        return LineResponse.of(line.updateInfo(lineRequest));
+    }
+
+    @Transactional
+    public void deleteById(Long id) {
+        checkExistLine(lineRepository.findById(id), id);
+        lineRepository.deleteById(id);
+    }
+
+    private Line getLine(Long id) {
+        Optional<Line> line = lineRepository.findById(id);
+        checkExistLine(line, id);
+        return line.get();
```

Comment:
```suggestion
        return line.orElseThrow(() -> new EntityNotFoundException("Line", id));
```
메서드로 분리한것도 좋지만 이렇게 해도 나쁘지 않을것 같아요!